pyiron_contrib/generic/s3io.py

```diff
@@ -0,0 +1,567 @@
+        return groups
+
+    def list_nodes(self):
+        """
+        List of 'files' ( string not followed by '/' ) in the current group.
+
+        Returns:
+            list: list of file names.
+        """
+        nodes = []
+        for obj in self._bucket.objects.filter(Prefix=self._bucket_path, Delimiter='/'):
+            nodes.append(obj.key.split('/')[-1])
+        return nodes
+
+    def list_all(self):
+        """
+        Combination of list_groups() and list_nodes() in one dictionary with the corresponding keys:
+        - 'groups': Sub-folder/ -groups.
+        - 'nodes': Files in the current group.
+
+        Returns:
+            dict: dictionary with all items in the group.
+        """
+        return {
+            "groups": self.list_groups(),
+            "nodes": self.list_nodes(),
+        }
+
+    def _to_abs_bucketpath(self, path):
+        """Helper function to convert a given path to an absolute path inside the S3 bucket."""
+        if path is None or "":
+            path = self._bucket_path
+        if posixpath.isabs(path):
+            path = path[1:]
+        else:
+            path = self._bucket_path + path
+        return path
+
+    def is_dir(self, path):
+        """
+        Check if given path is a directory.
+
+        Args:
+            path (str): path to check.
+
+        Returns:
+            bool: True if path is a directory.
+        """
+        path = self._to_abs_bucketpath(path)
+        if len(path) > 1 and path[-1] != '/':
+            path = path + '/'
+        objs = list(self._bucket.objects.filter(Prefix=path))
+        return len(objs) > 0
+
+    def is_file(self, path):
+        """
+        Check if given path is a file.
+
+        Args:
+            path (str): path to check.
+
+        Returns:
+            bool: True if path is a file.
+        """
+        path = self._to_abs_bucketpath(path)
+        objs = list(self._bucket.objects.filter(Prefix=path))
+        for obj in objs:
+            if obj.key == path:
+                return True
+        return False
+
+    def open(self, group):
+        """
+        Opens the provided group (create group if not yet present).
+
+        Args:
+            group (str): group to open/create.
+        """
+        new = self.copy()
+        new.s3_path = self.s3_path + group
+        new.history.append(new.s3_path)
+        return new
+
+    def copy(self):
+        """
+        Copy the Python object which links to the S3 object store.
+
+        Returns:
+            FileS3IO: New FileS3io object pointing to the same S3 object store
+        """
+        new = FileS3IO(config=self._s3io, path=self.s3_path)
+        return new
+
+    def close(self):
+        """Close current group and open previous group."""
+        if len(self.history) > 1:
+            del self.history[-1]
+        elif len(self.history) == 1:
+            self.history[0] = "/"
+        else:
+            print("Err: no history")
```

Comment:
Could be silent or raise an error?  I'm not sure what the HDF5 equivalent does, so I'm open.

Reply:
It is silent, I also silenced here.